pkg/types/powervs/validation/machinepool.go

```diff
@@ -0,0 +1,69 @@
+package validation
+
+import (
+	"math"
+	"regexp"
+	"strconv"
+
+	"github.com/google/uuid"
+	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateMachinePool checks that the specified machine pool is valid.
+func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate VolumeIDs
+	for i, volumeID := range p.VolumeIDs {
+		_, err := uuid.Parse(volumeID)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeIDs").Index(i), volumeID, "volume ID must be a valid UUID"))
+		}
+	}
+
+	// Validate Memory
+	if p.Memory != "" {
+		memory, err := strconv.ParseInt(p.Memory, 10, 64)
+		if err == nil {
+			if memory < 2 || memory > 64 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
+			}
+		} else {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
+		}
+	}
+
+	// Validate Processors
+	if p.Processors != "" {
+		processors, err := strconv.ParseFloat(p.Processors, 64)
+		if err == nil {
+			if processors < 0.25 || processors > 32 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "number of processors must be from .25 to 32 cores"))
+			}
+			if math.Mod(processors*1000, 2) != 0 || math.Mod(processors*100, 25) != 0 {
```

Comment:
This is still problematic with floating-point arithmetic. Multiplying does not change anything with regards to precision.

Reply:
can you give me a sample input for which this would be problematic? all the test inputs i've given have worked so i'm missing something

Reply:
There are no examples of it failing for the valid values since floating points have binary precision and we are dealing with quarters. However, this is fragile and easily avoided by using regex matching of the string instead of floating point arithmetic.